components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/builders/ResponseBuilder.java

```diff
@@ -17,12 +17,23 @@
  */
 package org.wso2.carbon.identity.sso.saml.builders;
 
+import org.joda.time.DateTime;
 import org.opensaml.saml2.core.Response;
 import org.wso2.carbon.identity.base.IdentityException;
 import org.wso2.carbon.identity.sso.saml.dto.SAMLSSOAuthnReqDTO;
 
+import java.security.NoSuchAlgorithmException;
+
 public interface ResponseBuilder {
 
-    Response buildResponse(SAMLSSOAuthnReqDTO authnReqDTO, String sessionIndexId) throws IdentityException;
+    @Deprecated
```

Comment:
When we deprecate something, we need to comment why it is deprecated and what is the correct alternative.